flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandler.java

```diff
@@ -193,6 +199,16 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object msg) throws Exc
 			if (triggerWrite) {
 				writeAndFlushNextMessageIfPossible(ctx.channel());
 			}
+		} else if (msg instanceof ResumeConsumptionEvent) {
+			RemoteInputChannel inputChannel = ((ResumeConsumptionEvent) msg).inputChannel;
+			inputChannel.unblockChannel();
+			ResumeConsumption resumeConsumption = new ResumeConsumption(inputChannel.getInputChannelId());
+
+			ctx.channel().writeAndFlush(resumeConsumption).addListener((ChannelFutureListener) channelFuture -> {
```

Comment:
It should make use of existing `writeAndFlushNextMessageIfPossible` for writing any messages on downstream side, otherwise we might miss some conditions such as `if (channelError.get() != null || !channel.isWritable()) ` inside previous `writeAndFlushNextMessageIfPossible`. And also bring much overhead work for maintaining two different paths.

We can refactor the existing `inputChannelsWithCredit` as a more general outbound message queue to insert both `AddCredit` and `ResumeConsumption` messages. And define an abstract `ClientOutboundMessage` with `buildMessage` method to be implemented by `AddCredit` and `ResumeConsumption` separately during `writeAndFlushNextMessageIfPossible`.